### DISCUSSION TITLE
making remark-math more useful than it is right now (at least as a standalone package)

### DISCUSSION THREAD

**Discussion by karuifeather** [remarkjs/remark#698] [Ideas] [answered]
Hello!

Here is an example md.

```markdown
$ x + 1 = 2 $. 

$$
x + 1 = 2
$$
```

Wouldn't it be great if `remark-math` gave this output instead of what it produces right now?

```html
<p><span class="math math-inline">\(x + 1 = 2 \)</span></p>
<div class="math math-display">\[x + 1 = 2\]</div>
```

Including delimiters in the output html will make remark-math more useful as a stand alone package than it is right now. Yes, [Pandoc](https://pandoc.org) produces similar result.

If this is done, users can easily use Katex client-side scripts to render Math on browsers without the need of `rehype-katex`. One less dependency and less html code to send between each requests.

Back to topic: what I am proposing is giving users a choice to either client-side or server-side Katex/Mathjax because as of right now, to my understanding, if you want to have Math in markdown, then you have to use remark-math which again gets useful *only* when used together with server-side method of rendering Math i.e `rehype-katex `or `rehype-mathjax`.

**OR** Is there already some way to do this and I'm not looking at the right place? If so, please help this poor soul.

**Comment by wooorm** [comment, marked as answer]
`rehype-mathjax` has a browser option that does this?

> One less dependency and less html code to send between each requests.

Client side JavaScript is typically much heavier than HTML. Especially for bigger libraries like KaTeX and MathJax, the impact engines having to parse and run all that JavaScript is giant, compared to shipping HTML down the write.
Plus you’d still have a dependency, but in `<script>` tags.
